src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -412,13 +407,40 @@ public static void AddSelfLinks(HttpRequest request, Instance instance)
             {
                 foreach (DataElement dataElement in instance.Data)
                 {
-                    dataElement.SelfLinks ??= new ResourceLinks();
-
-                    dataElement.SelfLinks.Platform = $"{selfLink}/data/{dataElement.Id}";
+                    AddDataSelfLinks(selfLink, dataElement);
                 }
             }
         }
 
+        /// <summary>
+        /// Computes the self link (url) to an instance.
+        /// </summary>
+        /// <param name="request">the http request that has scheme, host and path properties</param>
+        /// <param name="instance">the instance which has the instance id</param>
+        /// <returns>A string that contains the self link url to the instance</returns>
+        public static string ComputeInstanceSelfLink(HttpRequest request, Instance instance)
+        {
+            string selfLink = $"{request.Scheme}://{request.Host.ToUriComponent()}{request.Path}";
```

Comment:
This might generate invalid links. One difficult part of having multiple infrastructure layers (Application gateway and API management) is that the actual host for the server generating this selfLink might be different from the host visible to external parties.

Reply:
I will make an issue to analyse and consider a better solution for this.